### PR TITLE
Fix code generation for effect chain

### LINKED ISSUE #116: Final expression in codegenChain not being run
This has to do with how codegenChain is currently defined. Reading through initially, this might have to do with how EffectPure is being unwrapped as a special case rather than being handled in the "final" step in the accumulation.

### PR DESCRIPTION
**Pull request by purefunctor**
Closes #116. Should finally unblock #109

### CODE REVIEW

File: src/PureScript/Backend/Chez/Convert.purs [hunk 1]
```diff
@@ -227,43 +227,51 @@ codegenLiteral codegenEnv = case _ of
...
         ]
 
+  collect :: Array _ -> NeutralExpr -> ChezExpr
+  collect bindings expression = case unwrap expression of
+    Let i l v e' ->
+      collect (Array.cons { i, l, v: codegenExpr codegenEnv v } bindings) e'
+    EffectPure e' | chainMode.effect ->
+      finish false bindings e'
+    EffectBind i l v e' | chainMode.effect ->
+      collect (Array.cons { i, l, v: S.chezUnthunk $ codegenExpr codegenEnv v } bindings) e'
```
Comment (JordanMartinez) [inline src/PureScript/Backend/Chez/Convert.purs:269]: Rather than consing here and reversing in `finish`, why not `snoc` so that reversing isn't needed?

Comment (purefunctor) [reply src/PureScript/Backend/Chez/Convert.purs:269]: Ah, `snoc` is technically cheaper than `cons` for arrays, right?

Comment (purefunctor) [reply src/PureScript/Backend/Chez/Convert.purs:269]: or well, I don't think it'd matter much in this specific instance. `snoc` is indeed much cleaner

**Review by JordanMartinez — APPROVED** [review]
I'm still not sure if this is the best way to codegen effect but I'm approving this so we can unblocked your other PR.

**Pull request merged**